static/app/views/alerts/rules/issue/ticketRuleModal.tsx

```diff
@@ -179,7 +179,14 @@ class TicketRuleModal extends AbstractExternalIssueForm<Props, State> {
   getErrors() {
     const errors: ExternalIssueFormErrors = {};
     for (const field of this.cleanFields()) {
-      if (field.type === 'select' && field.default) {
+      // If the field is a select and has a default value, make sure that the
+      // default value exists in the choices. Skip check if the default is not
+      // set, an empty string, or an empty array.
+      if (
+        field.type === 'select' &&
+        field.default &&
+        (!Array.isArray(field.default) || field.default.length > 0)
```

Comment:
You're checking to ensure that the default is _not_ an array, or that it has a length...

what if it's default value _is_ an array with a length?
Doesn't that negate the first check of this clause?

Could you instead frame this more like
```
// If the default is an empty array
if (Array.isArray(field.default) && !field.default.length)
```
?

Reply:
If it's an array with a length, we do want it to go into the if. I'll refactor it so the check goes from:

`(!Array.isArray(field.default) || field.default.length > 0)`
_not array OR not empty_
TO
`!(Array.isArray(field.default) && !field.default.length)`
_not(array and empty)_